faculty/clients/experiment.py

```diff
@@ -245,6 +330,93 @@ def make_restore_runs_response(self, data):
         return RestoreExperimentRunsResponse(**data)
 
 
+class SingleFilterValueField(fields.Field):
+    """
+    Field that serialises/deserialises a run filter.
+    """
+
+    def _is_valid_uuid(self, value, obj):
+        return isinstance(value, uuid.UUID) and (
+            obj.by in {SingleFilterBy.PROJECT_ID, SingleFilterBy.RUN_ID}
+        )
+
+    def _is_valid_experiment_id(self, value, obj):
+        return (
+            isinstance(value, int) and obj.by == SingleFilterBy.EXPERIMENT_ID
+        )
+
+    def _is_directly_stringifiable(self, value, obj):
+        return (
+            self._is_valid_uuid(value, obj)
+            or self._is_valid_experiment_id(value, obj)
+            or obj.by
+            in {
+                SingleFilterBy.TAG,
+                SingleFilterBy.PARAM,
+                SingleFilterBy.METRIC,
+            }
+        )
+
+    def _deserialize(self, value, attr, obj, **kwargs):
+        pass
```

Comment:
Not really sure what the sensible thing to do is here. This is strictly a request payload, which I don't predict we will need to deserialise any time soon.